pkg/streamingpromql/types/operator.go

```diff
@@ -50,15 +52,15 @@ type RangeVectorOperator interface {
 	// SeriesMetadata must be called exactly once before calling NextSeries.
 	NextSeries(ctx context.Context) error
 
-	// NextStepSamples populates the provided RingBuffer with the samples for the next time step for the
+	// NextStepSamples populates the provided RingBuffers with the samples for the next time step for the
 	// current series and returns the timestamps of the next time step, or returns EOS if no more time
 	// steps are available.
-	// The provided RingBuffer is expected to only contain points for the current series, and the same
-	// RingBuffer should be passed to subsequent NextStepSamples calls for the same series.
-	// The provided RingBuffer may be populated with points beyond the end of the expected time range, and
+	// The provided RingBuffers are expected to only contain points for the current series, and the same
+	// RingBuffers should be passed to subsequent NextStepSamples calls for the same series.
+	// The provided RingBuffers may be populated with points beyond the end of the expected time range, and
 	// callers should compare returned points' timestamps to the returned RangeVectorStepData.RangeEnd.
 	// Next must be called at least once before calling NextStepSamples.
-	NextStepSamples(floats *RingBuffer) (RangeVectorStepData, error)
+	NextStepSamples(floats *RingBuffer[promql.FPoint], histograms *RingBuffer[promql.HPoint]) (RangeVectorStepData, error)
```

Comment:
We probably need a warning here like the one you added for `InstantVectorSeriesData` mentioning that different `HPoint`s might all refer to the same `FloatHistogram`.